src/mscorlib/shared/System/Memory.cs

```diff
@@ -257,7 +257,7 @@ public bool TryGetArray(out ArraySegment<T> arraySegment)
                     return true;
                 }
             }
-            else
+            else if (_object != null)
             {
                 T[] arr = _object as T[];
                 if (typeof(T) != typeof(char) || arr != null)
```

Comment:
Rather than adding `if (_object != null)`, you could probably just simplify this down to:
```C#
else if (_object is T[] arr)
{
    arraySegment = new ArraySegment<T>(arr, _index, _length);
    return true;
}
```

Reply:
Do we still need the `if (typeof(T) != typeof(char)` check in this case?